pkg/lca/seedgenerator.go

```diff
@@ -0,0 +1,294 @@
+package lca
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	lcasgv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/seedgenerator/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SeedGenerator provides struct for the seedgenerator object containing connection to
+// the cluster and the seedgenerator definitions.
+type SeedGenerator struct {
+	// SeedGenerator definition. Used to store the seedgenerator object.
+	Definition *lcasgv1alpha1.SeedGenerator
+	// Created seedgenerator object.
+	Object *lcasgv1alpha1.SeedGenerator
+	// Used in functions that define or mutate the seedgenerator definition.
+	// errorMsg is processed before the seedgenerator object is created
+	errorMsg  string
+	apiClient *clients.Settings
+}
+
+// SeedGeneratorAdditionalOptions additional options for imagebasedupgrade object.
+type SeedGeneratorAdditionalOptions func(builder *SeedGenerator) (*SeedGenerator, error)
+
+// NewBuilder creates a new instance of SeedGenerator.
+func NewSeedGeneratorBuilder(
+	apiClient *clients.Settings,
+	name string,
+) *SeedGenerator {
+	builder := SeedGenerator{
+		apiClient: apiClient,
+		Definition: &lcasgv1alpha1.SeedGenerator{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the seedgenerator is empty")
+
+		builder.errorMsg = "SeedGenerator name cannot be empty"
+	}
+
+	return &builder
+}
+
+// WithOptionsSeedGenerator creates seedgenerator with generic mutation options.
+func (builder *SeedGenerator) WithOptionsSeedGenerator(options ...SeedGeneratorAdditionalOptions) *SeedGenerator {
```

Comment:
```suggestion
func (builder *SeedGenerator) WithOptions(options ...SeedGeneratorAdditionalOptions) *SeedGenerator {
```
This is a method do it doesn't have to be unique

Reply:
Updated. Thanks.